### PR TITLE
added deprecated helper

### PR DESCRIPTION
**Pull request by saeideng**
https://github.com/cakephp/cakephp/pull/11241#issuecomment-332301887

### CODE REVIEW

File: src/TestSuite/TestCase.php [hunk 1]
```diff
@@ -88,6 +88,20 @@ public function skipIf($shouldSkip, $message = '')
...
+     * Helper method for check deprecation methods
+     *
+     * @param callable $callable callable function that will receive asserts
+     * @return void
+     */
+    public function deprecated($callable)
+    {
+        $errorLevel = error_reporting();
+        error_reporting(E_ALL ^ E_USER_DEPRECATED);
+            $callable();
```
Comment (markstory) [inline src/TestSuite/TestCase.php]: The indentation here looks off.

**Review by markstory — COMMENTED** [review]
I like this approach. It helps contain the deprecated feature tests and will make them easier to remove later on.

**Comment by saeideng** [comment]
even we can assert exception message
But there is a bit problem
line and file path of exception message return 
`....src/.... TestSuite/testcase.php line 100 `
So we can not assert this line/file path
But I think we no needs to assert to message
That is  a bit hard to implement
But usage :

```php
This->deprecated(function(){
... 
},[
'Message'=>....
'MessageRegex'=>...
])
```
But Im not sure line after that can be works or no
I need a test

**Comment by saeideng** [comment]
Lines after 'deprecated()' is works correctly now (without assert message)

File: src/TestSuite/TestCase.php (hunk 1 above)
Comment (saeideng) [reply src/TestSuite/TestCase.php]: fixed

**Comment by saeideng** [comment]
you can merge this if no need to assert message
we can add extra assert in future if necessary

**Pull request merged**